state-chain/runtime/src/chainflip/address_derivation/sol.rs

```diff
@@ -0,0 +1,54 @@
+use cf_chains::{
+	address::AddressDerivationApi, assets::sol::Asset, sol::DerivedAddressBuilder, Solana,
+};
+
+use crate::Environment;
+
+use super::AddressDerivation;
+
+const VAULT_PDA_ASSET_SOL: &str = "VaultPdaAssetSol";
+
+impl AddressDerivationApi<Solana> for AddressDerivation {
+	fn generate_address(
+		source_asset: <Solana as cf_chains::Chain>::ChainAsset,
+		channel_id: cf_primitives::ChannelId,
+	) -> Result<
+		<Solana as cf_chains::Chain>::ChainAccount,
+		cf_chains::address::AddressDerivationError,
+	> {
+		let (address, _) = <Self as AddressDerivationApi<Solana>>::generate_address_and_state(
+			source_asset,
+			channel_id,
+		)?;
+		Ok(address)
+	}
+
+	fn generate_address_and_state(
+		source_asset: <Solana as cf_chains::Chain>::ChainAsset,
+		channel_id: cf_primitives::ChannelId,
+	) -> Result<
+		(
+			<Solana as cf_chains::Chain>::ChainAccount,
+			<Solana as cf_chains::Chain>::DepositChannelState,
+		),
+		cf_chains::address::AddressDerivationError,
+	> {
+		let vault_address = Environment::sol_vault_address();
+		match source_asset {
+			Asset::Sol => {
+				let (pda, _bump) = DerivedAddressBuilder::from_address(vault_address)?
+					.chain_seed(VAULT_PDA_ASSET_SOL)?
```

Comment:
We have not decided yet on the best/default way to generate deposit addresses but this seed (`VAULT_PDA_ASSET_SOL`) is unnecessary, we can just derive it from the `channel_id`. We will maybe tweak this in the future but for now I'd remove it, as afaik there is no reason to do it and it will cost us more fees for no reason.

Reply:
done!